tests/conftest.py

```diff
@@ -4,10 +4,13 @@
 
 def pytest_addoption(parser):
     parser.addoption("--package", action="store")
+    parser.addoption("--names-to-skip", action="store", default="")
 
 def pytest_generate_tests(metafunc):
```

Comment:
I think that this way of parameterizing a single function is very odd. You are creating test cases for every object that is the output of `traverse_package` with the global conftest.py file. 

Why not parameterize the relevant test (`test_has_docstring`) directly?

Reply:
It's helpful to have a separate parameterized test case for each object so that you can see if multiple objects are missing docstrings.

By parameterizing using conftest.py I can pass in the package as an argument, which is important to make the test reusable for different packages.

Reply:
I am not arguing against parameterization, this is good. I am arguing against using global parameterization with the conftest (and then filter) instead of parameterizing the function directly.

Reply:
This is my source: https://stackoverflow.com/questions/40880259/how-to-pass-arguments-in-pytest-by-command-line

Also see here: https://docs.pytest.org/en/latest/how-to/parametrize.html#basic-pytest-generate-tests-example

Using metafunc in this way seems like a standard way to dynamically parameterize tests.

Reply:
I don't think you got me. I am not saying that it does not work. I am saying that you are killing a fly with a bazooka and that using conftest.py will make it harder to pack the thing as an action.

You can do both parsing config all those things in the top of the `test_has_docstring` test or in the same file.. You don't really need to mess with global pytest config (conftest.py).

Let me put something together.

Reply:
Ah, forget about this. I was wrongly thinking that `conftest.py` needed to be global but there can be one per folder so my whole point is moot.